org.scala-ide.sdt.core/src/org/scalaide/core/ScalaIdeDataStore.scala

```diff
@@ -0,0 +1,57 @@
+package org.scalaide.core
+
+import java.io.File
+import java.io.IOException
+
+import org.scalaide.util.eclipse.EclipseUtils
+
+object ScalaIdeDataStore {
+
+  private val userHome = System.getProperty("user.home")
+  private val sep = File.separator
+
+  /**
+   * Preference store id that points to the location of Scala IDEs own Eclipse
+   * independent data store.
+   */
+  final val DataStoreId = "org.scalaide.core.dataStore"
+
+  /**
+   * The default location of the data store. Users can change it in the Scala
+   * preference page. The data store can be shared between multiple Eclipse
+   * instances.
+   */
+  final val DefaultDataStoreLocation = s"$userHome$sep.scalaide"
+
+  /**
+   * The absolute path to the data store.
+   */
+  def dataStoreLocation: String =
+    IScalaPlugin().getPreferenceStore.getString(DataStoreId)
+
+  def statisticsLocation: String =
+    s"$dataStoreLocation${sep}statistics"
```

Comment:
For String concatenation I prefer `+`, imo `dataStoreLocation + sep + "statistics"` is more readable. But nitpicking.